modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/util/JournalUtil.java

```diff
@@ -239,17 +238,19 @@ public static long getPreviewPlid(
 		}
 
 		PermissionChecker permissionChecker =
-			PermissionCheckerFactoryUtil.create(themeDisplay.getUser());
+			themeDisplay.getPermissionChecker();
 
 		if (!LayoutPermissionUtil.contains(
 				permissionChecker, layout, ActionKeys.VIEW)) {
 
+			Long firstLayoutId = layout.getPlid();
+
 			layout = _partialSearchLayout(
-				themeDisplay, permissionChecker, false);
+				themeDisplay, permissionChecker, firstLayoutId, false);
 
 			if (layout == null) {
 				layout = _partialSearchLayout(
-					themeDisplay, permissionChecker, true);
+					themeDisplay, permissionChecker, firstLayoutId, true);
 			}
 		}
```

Comment:
I don't think ignoring the first Layout by checking id is a good idea.
May change [these](https://github.com/tototrinh/liferay-portal/blob/84be4808a650d82063b34d5f36a9cc2b1e857439/modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/util/JournalUtil.java#L230-L255) to something like:
```
Layout layout = _partialSearchLayout(themeDisplay, false);

if (layout == null) {
	layout = _partialSearchLayout(themeDisplay, true);
}
```
btw, I think `_partialSearchLayout` doesn't make sense for the purpose of your function. Thanks.

Reply:
Thanks for your review. I understand your idea. But: I think the issue is the special scenario. If we replace completely `fetchFirstLayout` by `getLayouts` we'll lose performance for normal cases. The "fetch one" method is always better than the "get list" method.

Reply:
I see. Thanks for letting me know. In this case, I think we don't need to ignore the first layout. it's not a big deal.
Please help change the name of the function to make it more clear. Thanks.

Reply:
I think we don't need to set permissionChecker as a parameter of the function. Using themeDisplay.getPermissionChecker() directly

Reply:
Ok, I'll modify code. Thanks a lot.